CogniteSdk.Types/Beta/DataModels/Instances/Sourced.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace CogniteSdk.Beta.DataModels
+{
+    /// <summary>
+    /// Special type for instances with data for a specific source,
+    /// to be used with `BaseDataModelResource`.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public abstract class SourcedInstanceWrite<T>
```

Comment:
This is almost a base class for `SourcedInstance<T>`, would be better to split off one or more base classes.